src/main/webapp/group.js

```diff
@@ -22,17 +22,17 @@ function addEventListeners() {
 
 /** A function that displays information about a group at the top of the group's page. */
 function loadGroupHeader() {
-    loadGroupHeaderHelper(document, window);
+    return loadGroupHeaderHelper(document, window);
 }
 
-function loadGroupHeaderHelper(document, window) {
+async function loadGroupHeaderHelper(document, window) {
     const params = new URLSearchParams();
 
     var queryString = new Array();
     window.onload = readComponents(queryString, window);
     const groupId = queryString["groupId"];
 
-    fetch("/group-data?groupId=" + groupId).then(response => response.json()).then((result) => {
+    await fetch("/group-data?groupId=" + groupId).then(response => response.json()).then((result) => {
```

Comment:
Why is there a need to await now?

Reply:
This is mainly for the tests. The container wasn't being updated in the tests so I had to add an await to make sure the fetch call finished before I tested it.